zashi-pipeline.py

```diff
@@ -0,0 +1,217 @@
+        font-size: 14px;
+        line-height: 1.5;
+        word-wrap: break-word;
+      }
+      a {
+        color: #0969da;
+      }
+      table {
+        border-collapse: collapse;
+        width: 100%;
+        width: max-content;
+        max-width: 100%;
+        overflow: auto;
+      }
+      table tr {
+        border-top: 1px solid #d1d9e0b3;
+      }
+      table th, table td {
+        border: 1px solid #d1d9e0b3;
+        padding: 6px 13px;
+        text-align: center;
+      }
+
+      @media (prefers-color-scheme: dark) {
+        body {
+          background-color: #121212;
+          color: #f0f6fc;
+        }
+        a {
+          color: #4493f8;
+        }
+        table tr {
+            border-top: 1px solid #3d444db3;
+        }
+        table th, table td {
+            border: 1px solid #3d444db3;
+        }
+      }
+    </style>
+  </head>
+  <body>
+    <h1>Zashi Pipeline</h1>
+    <p>🐞 = bug, 💡 = feature, ✅ = implemented / released, 🛑 = unfinished, 📥 = untracked</p>
+    <table>
+      <thead>
+        <tr>
+          <th>Type</th>
+          <th>Issue</th>
+          <th>Rust crate</th>
+          <th>Android SDK</th>
+          <th>Swift SDK</th>
+          <th>Zashi Android</th>
+          <th>Zashi iOS</th>
+        </tr>
+      </thead>
+      <tbody>
+'''
+    html_footer = '''
+      </tbody>
+    </table>
+  </body>
+</html>
+'''
+    with open('public/zcash-change-deployment-pipeline.html', 'w') as f:
+        f.write(html_header)
+
+        for issue in tracked_issues.values():
+            f.write('<tr>')
+
+            if 'C-tracked-bug' in issue.labels:
+                f.write('<td>🐞</td>')
+            else:
+                f.write('<td>💡</td>')
+
+            f.write('<td>{} <a href="{}">{}</a></td>'.format(
+                '✅' if issue.state == 'closed' else '🛑',
+                issue.url,
+                issue.title,
+            ))
+
+            children = nx.descendants(dg, issue)
+            def find_child_release(repo_id):
+                for child in children:
+                    if child.repo_id == repo_id and 'C-release' in child.labels:
+                        # Extract version number from title
+                        if repo_id == RUST:
+                            version = re.search(r'zcash_[^ ]+ \d+(\.\d+)+', child.title).group()
+                        else:
+                            version = re.search(r'\d+(\.\d+)+', child.title).group()
+
+                        f.write('<td>{} <a href="{}">{}</a></td>'.format(
+                            '✅' if child.state == 'closed' else '🛑',
+                            child.url,
+                            version,
+                        ))
+                        return
+
+                # Release not found in this repo
+                f.write('<td>📥</td>')
+
+            find_child_release(RUST)
```

Comment:
Current limitation: we only render a row for the first Rust crate release issue we find. That is most likely the one with the smallest issue number.